src/tiled/tileseteditor.cpp

```diff
@@ -435,8 +444,35 @@ void TilesetEditor::addTiles()
         QPixmap image;
     };
     QVector<LoadedFile> loadedFiles;
-
+    // If the tile is already in the tileset, warn user and confirm addition
+    bool dontAskAgain = false;
+    bool rememberOption = true;
     for (const QString &file : files) {
+        bool addImage = true;
+        if (hasTileInTileset(file, mCurrentTilesetDocument->tileset().data()->tiles())) {
+            if (dontAskAgain) 
+                addImage = rememberOption;
```

Comment:
Now you can just `continue;` here, because if `dontAskAgain` you know that `rememberOption == false`.